aten/src/ATen/native/Linear.cpp

```diff
@@ -825,6 +826,14 @@ Tensor tensordot(const Tensor& input1, const Tensor& input2, IntArrayRef dims1,
 }
 
 Tensor &tensordot_out(const Tensor& input1, const Tensor& input2, IntArrayRef dims1, IntArrayRef dims2, Tensor& result) {
+  if(result.defined()) {
```

Comment:
Can you move this check to where the other ones are below and only error if the shape of the user provided out tensor and the result_tmp are different?

Reply:
Hi @soulitzer , thanks for the quick response.

My intuition is to always prevent inplace memory operations for out tensor under requries_grad. May I know the reson for having error out only when shape mismatch, and discarding this quick error out?

Reply:
@cz2h  In-place operations are generally supported under autograd., e.g. `add_`. out= indeed doesn't support autograd today, but I don't think there's a fundamental reason why it cannot.

Reply:
Sorry for delay and thanks for the example. 

That's a good point - the "bc-breaking"-ness with the current PR are (1) if the user ends up never calling backward they would never error (2) if the user did not have requires_grad=True for the out= tensor things are fine even if they call backward and the shape is resized.

There's no way to avoid (1). So let's not worry about that, also the user can easily work around this by detaching. 
For (2), we can limit the error to the particular case when the out= variable requires_grad (and grad_mode is True)

Reply:
> Sorry for delay @cz2h and thanks for the example.
> 
> That's a good point - the "bc-breaking"-ness with the current PR are (1) if the user ends up never calling backward they would never error (2) if the user did not have requires_grad=True for the out= tensor things are fine even if they call backward and the shape is resized.
> 
> There's no way to avoid (1). So let's not worry about that, also the user can easily work around this by detaching. For (2), we can limit the error to the particular case when the out= variable requires_grad (and grad_mode is True)

Thanks for the investigation, the use case is clear to me.

I have narrowed the error case in latest commit please feel free to take a look. Thanks in advanced.